subsys/bluetooth/host/adv.c

```diff
@@ -1332,6 +1357,20 @@ int bt_le_ext_adv_delete(struct bt_le_ext_adv *adv)
 #endif /* defined(CONFIG_BT_EXT_ADV) */
 
 
+static void adv_timeout(struct k_work *work)
+{
+#if defined(CONFIG_BT_EXT_ADV)
+	struct bt_le_ext_adv *adv = CONTAINER_OF(work, struct bt_le_ext_adv,
+	timeout_work);
+
+	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
+		bt_le_ext_adv_stop(adv);
+	}
+#else
+	bt_le_adv_stop();
```

Comment:
I think we should add a check for the error code here, and at log if there's an error, and possibly add an __ASSERT as that should never happen.